app/controllers/application_controller.rb

```diff
@@ -217,26 +206,21 @@ def fix_broken_personal_key_url
   end
 
   def after_sign_in_path_for(_user)
-    accept_rules_of_use_url ||
-      user_suspended_url ||
-      service_provider_mfa_setup_url ||
-      add_piv_cac_setup_url ||
-      fix_broken_personal_key_url ||
-      user_session.delete(:stored_location) ||
-      sp_session_request_url_with_updated_params ||
-      signed_in_url
-  end
-
-  def signed_in_url
-    return user_two_factor_authentication_url unless user_fully_authenticated?
+    return rules_of_use_path if !current_user.accepted_rules_of_use_still_valid?
+    return user_please_call_url if current_user.suspended?
+    return authentication_methods_setup_url if user_needs_sp_auth_method_setup?
+    return login_add_piv_cac_prompt_url if session[:needs_to_setup_piv_cac_after_sign_in].present?
+    return fix_broken_personal_key_url if current_user.broken_personal_key?
+    return user_session.delete(:stored_location) if user_session.key?(:stored_location)
     return reactivate_account_url if user_needs_to_reactivate_account?
-    return url_for_pending_profile_reason if user_has_pending_profile?
     return backup_code_reminder_url if user_needs_backup_code_reminder?
```

Comment:
My instinct with moving this from `signed_in_url` to `after_signed_in_path_for` is that we want to show the backup code reminder to everyone signing in (i.e. through a partner application), not just those who are coming to Login.gov directly.

However, in testing this, I noticed that the acknowledgement on the reminder screen "I have my codes" is hard-coded to redirect the user to their account page, _not_ to continue them to the next step in a partner authentication.

https://github.com/18F/identity-idp/blob/885af33d951d2e1d237123a27c4a481bb5993302/app/views/users/backup_code_setup/reminder.html.erb#L11-L17

So what I'm going to do for now is:

- Return this to the `signed_in_url` logic, so that it's only considered when visiting Login.gov directly.
- Create a ticket to reconsider the backup code reminder to show for all authentications, where acknowledgment should continue them in the next step of a partner authentication.

Reply:
Updated in bd2d093.

Ticket at [LG-10965](https://cm-jira.usa.gov/browse/LG-10965)